omx/gstomxaudiodec.c

```diff
@@ -867,29 +850,38 @@ gst_omx_audio_dec_set_format (GstAudioDecoder * decoder, GstCaps * caps)
     GST_DEBUG_OBJECT (self, "Decoder drained and disabled");
   }
 
+  /* Set up audiodec pcm parameters */
+  GST_OMX_INIT_STRUCT (&self->pcm_param);
+  self->pcm_param.nPortIndex = self->dec_out_port->index;
+  self->pcm_param.ePCMMode = OMX_AUDIO_PCMModeLinear;
+  self->pcm_param.bInterleaved = OMX_TRUE;
+  self->pcm_param.nChannels = 2;
+  self->pcm_param.eChannelMapping[0] = OMX_AUDIO_ChannelLF;
+  self->pcm_param.eChannelMapping[1] = OMX_AUDIO_ChannelRF;
+  self->pcm_param.eNumData = OMX_NumericalDataSigned;
+  self->pcm_param.eEndian = OMX_EndianLittle;
+  self->pcm_param.nBitPerSample = 16;
+  /* We get the sampling rate from caps */
+  self->pcm_param.nSamplingRate = 0;
+
+  /* Call subclass set_format for input port configuration */
   if (klass->set_format) {
     if (!klass->set_format (self, self->dec_in_port, caps)) {
-      GST_ERROR_OBJECT (self, "Subclass failed to set the new format");
+      GST_ERROR_OBJECT (self, "Subclass failed to set the new input format");
       return FALSE;
     }
   }
 
-  GST_DEBUG_OBJECT (self, "Updating outport port definition");
-  if (gst_omx_port_update_port_definition (self->dec_out_port,
-          NULL) != OMX_ErrorNone)
-    return FALSE;
+  /* Call subclass set_format for output port configuration */
+  if (klass->set_format) {
```

Comment:
This conditional is unnecessary, move it to the `if (klass->set_format)` above

Reply:
Removed: de21a4ad7bcb37241fc66178972f27febe8b83f5